src/www/app/server/access_config.ts

```diff
@@ -0,0 +1,175 @@
+// Copyright 2018 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {makeConfig, SHADOWSOCKS_URI, SIP002_URI} from 'ShadowsocksConfig';
+import {ServerUrlInvalid, ServerIncompatible, ShadowsocksUnsupportedCipher} from '../../model/errors';
+import {ShadowsocksConfig} from '../config';
+
+export enum OutlineServerSupportedCipher {
+  CHACHA20_IETF_POLY1305 = 'chacha20-ietf-poly1305',
+  AES_128_GCM = 'aes-128-gcm',
+  AES_192_GCM = 'aes-192-gcm',
+  AES_256_GCM = 'aes-256-gcm',
+}
+
+export enum OutlineServerAccessType {
+  SHADOWSOCKS_URI,
+  SHADOWSOCKS_CONFIG,
+}
+
+export class OutlineServerAccessConfig implements ShadowsocksConfig {
```

Comment:
I was hoping to have a `AccessKey` type, with Dynamic and Static subtypes.

Reply:
but if the AccessKey can't have the config on it, then it's just a string